src/CleanSolutionTemplate.Domain/Common/Entity.cs

```diff
@@ -4,29 +4,17 @@ namespace CleanSolutionTemplate.Domain.Common;
 
 public abstract class Entity
 {
-    private readonly List<DomainEvent> _domainEvents = new();
+    readonly List<DomainEvent> _domainEvents = new();
 
-    protected Entity(Guid id)
-    {
-        this.Id = id;
-    }
+    protected Entity(Guid id) => this.Id = id;
 
     public Guid Id { get; private set; }
 
-    [NotMapped] public IEnumerable<DomainEvent> DomainEvents => this._domainEvents.AsReadOnly();
+    [NotMapped] public IReadOnlyList<DomainEvent> DomainEvents => this._domainEvents.AsReadOnly();
 
-    public void AddDomainEvent(DomainEvent domainEvent)
-    {
-        this._domainEvents.Add(domainEvent);
-    }
+    public void AddDomainEvent(DomainEvent domainEvent) => this._domainEvents.Add(domainEvent);
 
-    public void RemoveDomainEvent(DomainEvent domainEvent)
-    {
-        this._domainEvents.Remove(domainEvent);
-    }
+    public void RemoveDomainEvent(DomainEvent domainEvent) => this._domainEvents.Remove(domainEvent);
 
-    public void ClearDomainEvents()
-    {
-        this._domainEvents.Clear();
-    }
+    public void RemoveRange(int index) => this._domainEvents.RemoveRange(0, index);
```

Comment:
```suggestion
    public void ClearDomainEvents(int index) => this._domainEvents.Clear();
```

Let's leave the ClearDomainEvents for now, I see why you suggested this but I have a different idea on how to approach this. Will be implemented later.

Nice catch on the arrow functions and the IReadOnlyList instead of the IEnumerable